crates/valence_client/src/world_border.rs

```diff
@@ -0,0 +1,373 @@
+            } else {
+                (diameter.0, 0)
+            };
+
+            client.write_packet(&WorldBorderInitializeS2c {
+                x: c.0.x,
+                z: c.0.y,
+                old_diameter: diameter.0,
+                new_diameter,
+                portal_teleport_boundary: VarInt(ptb.0),
+                speed: VarLong(speed),
+                warning_blocks: VarInt(wb.0),
+                warning_time: VarInt(wt.0),
+            });
+        }
+    }
+}
+
+fn handle_diameter_change(
+    mut wbs: Query<(&mut Instance, &MovingWorldBorder), Changed<MovingWorldBorder>>,
+) {
+    for (mut ins, lerping) in wbs.iter_mut() {
+        if lerping.duration == 0 {
+            ins.write_packet(&WorldBorderSizeChangedS2c {
+                diameter: lerping.new_diameter,
+            })
+        } else {
+            ins.write_packet(&WorldBorderInterpolateSizeS2c {
+                old_diameter: lerping.current_diameter(),
+                new_diameter: lerping.new_diameter,
+                speed: VarLong(lerping.current_duration()),
+            });
+        }
+    }
+}
+
+fn handle_lerp_transition(mut wbs: Query<(&mut WorldBorderDiameter, &MovingWorldBorder)>) {
+    for (mut diameter, moving_wb) in wbs.iter_mut() {
+        if diameter.0 != moving_wb.new_diameter {
+            diameter.0 = moving_wb.current_diameter();
+        }
+    }
+}
+
+fn handle_center_change(
+    mut wbs: Query<(&mut Instance, &WorldBorderCenter), Changed<WorldBorderCenter>>,
+) {
+    for (mut ins, center) in wbs.iter_mut() {
+        ins.write_packet(&WorldBorderCenterChangedS2c {
+            x_pos: center.0.x,
+            z_pos: center.0.y,
+        })
+    }
+}
+
+fn handle_warn_time_change(
+    mut wb_query: Query<(&mut Instance, &WorldBorderWarnTime), Changed<WorldBorderWarnTime>>,
+) {
+    for (mut ins, wt) in wb_query.iter_mut() {
+        ins.write_packet(&WorldBorderWarningTimeChangedS2c {
+            warning_time: VarInt(wt.0),
+        })
+    }
+}
+
+fn handle_warn_blocks_change(
+    mut wb_query: Query<(&mut Instance, &WorldBorderWarnBlocks), Changed<WorldBorderWarnBlocks>>,
+) {
+    for (mut ins, wb) in wb_query.iter_mut() {
+        ins.write_packet(&WorldBorderWarningBlocksChangedS2c {
+            warning_blocks: VarInt(wb.0),
+        })
+    }
+}
+
+fn handle_portal_teleport_bounary_change(
+    mut wbs: Query<(
+        &mut Instance,
+        &WorldBorderCenter,
+        &WorldBorderWarnTime,
+        &WorldBorderWarnBlocks,
+        &WorldBorderDiameter,
+        &WorldBorderPortalTpBoundary,
+        Option<&MovingWorldBorder>,
+    )>,
+) {
+    for (mut ins, c, wt, wb, diameter, ptb, wbl) in wbs.iter_mut() {
+        let (new_diameter, speed) = if let Some(lerping) = wbl {
+            (lerping.new_diameter, lerping.current_duration())
+        } else {
+            (diameter.0, 0)
+        };
+
+        ins.write_packet(&WorldBorderInitializeS2c {
```

Comment:
This packet is being sent every tick. Did you forget to add `Changed<WorldBorderTpBoundary>`?